design/client-side-encryption_design.md

```diff
@@ -0,0 +1,554 @@
+# Proposal for client-side encryption
+
+## Abstract
+
+Currently only PersistentVolumes are stored in a Restic/Kopia repository, which means that only they are encrypted and the rest of the backup is not.
+It is desired that the cluster backup can be encrypted which overall will lead to better security.
+
+## Background
+
+As of today, Velero stores only PersistentVolume data in a Restic/Kopia repository.
```

Comment:
Can you perhaps explain how the Kopia repository is encrypted today. I am personally not aware of the mechanism, will be good to callout in this doc somewhere.

Reply:
There's a secret `velero-repo-credentials` containing the repository password. Per default, the password there has the value `static-passw0rd`. As you see, this is not a strong password and my guess is that the only reason this was used at all is because Kopia and Restic require encryption. While it is possible to change the password in the secret, it is as of yet not a documented feature.

Reply:
Documenting the secret is being tracked in #6443.

Reply:
Kopia encryption is described here: https://kopia.io/docs/advanced/encryption/.

Reply:
Thanks for explaining @jelemux / and the info @draghuram